dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -147,7 +148,10 @@ public sealed class SKFunction : ISKFunction, IDisposable
             {
                 string prompt = await functionConfig.PromptTemplate.RenderAsync(context).ConfigureAwait(false);
 
-                string completion = await client.CompleteAsync(prompt, requestSettings, context.CancellationToken).ConfigureAwait(false);
+                var completionResults = await client.GetCompletionsAsync(prompt, requestSettings, context.CancellationToken).ConfigureAwait(false);
+                string completion = await GetCompletionsResultContentAsync(completionResults, context.CancellationToken).ConfigureAwait(false);
+
+                context.LastPromptResults = completionResults.Select(c => c.ResultData).ToArray();
```

Comment:
Eeek, so this will mean that consumers will need to know that the `object` is an array? The `LastXYZ` design definitely falls over with SK pipelines.